metricflow/test/model/validations/test_measures.py

```diff
@@ -221,3 +223,33 @@ def test_reused_measure_alias_within_metric(simple_model__pre_transforms: UserCo
     assert (
         actual_error.find(expected_error_substring) != -1
     ), f"Expected error {expected_error_substring} not found in error string! Instead got {actual_error}"
+
+
+def test_invalid_non_additive_dimension_properties(simple_model__pre_transforms: UserConfiguredModel) -> None:
+    """Tests validator for invalid cases of non_additive_dimension properties."""
+    model = copy.deepcopy(simple_model__pre_transforms)
+    invalid_measure = Measure(
+        name="bad_measure",
+        agg=AggregationType.SUM,
+        non_additive_dimension=NonAdditiveDimensionParameters(
+            name="doesntexist",  # Time dimension doesn't exist
+            window_choice=AggregationType.SUM,  # Should only be MIN/MAX
+            window_groupings=["wherethisidentifier"],  # Identifier doesn't exist
+        ),
+        agg_time_dimension="ds",
+    )
+    model.data_sources[0].measures.append(invalid_measure)  # type: ignore
+    build = ModelValidator().validate_model(model)
+
+    assert len(build.issues.errors) == 3
+    expected_error_substring_1 = (
+        f"that is not defined in data source '{model.data_sources[0].name}' as a time dimension."
+    )
+    expected_error_substring_2 = "has a non_additive_dimension with an invalid 'window_choice'"
+    expected_error_substring_3 = "has a non_additive_dimension with an invalid 'window_groupings'"
+    for expected_str in [expected_error_substring_1, expected_error_substring_2, expected_error_substring_3]:
+        exist = False
+        for actual_str in build.issues.errors:
+            if actual_str.as_readable_str().find(expected_str) != -1:
+                exist = True
+        assert exist, f"Expected error {expected_str} not found in error string! Instead got {actual_str}"
```

Comment:
I think the error message on this last assertion only prints out the last actual_str value you happen to have seen.

I suggest something like this - I find it a little nicer to read and debug (well, once the formatting gets fixed up, anyway)

```suggestion
    missing_error_strings: Set[str] = {}
    for expected_str in [expected_error_substring_1, expected_error_substring_2, expected_error_substring_3]:
        if not any(actual_str.as_readable_str().find(expected_str) != -1 for actual_str in build.issues.errors):
            missing_error_strings.add(expected_str)

    assert len(missing_error_strings) == 0, f"Failed to match one or more expected errors: {missing_error_strings} in {set([x.as_readable_string() for x in build.issues.errors])}"
```